### PR TITLE
feat(rpc): add `blockTimestamp` to Log

### LINKED ISSUE #422: [Feature] Add `blockTimestamp` field to Log type
### Component

rpc

### Describe the feature you would like

Unblock [6752](https://github.com/paradigmxyz/reth/issues/6752) from reth:

> The proposed enhancement to include a blockTimestamp field in the eth_getLogs response is a pragmatic approach to solving a significant efficiency problem faced by Ethereum applications. It streamlines the process of fetching event data with timestamps, which is critical for many blockchain-based applications.

Potential breaking change if the field is not optional.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cairoeth**
## Motivation

Fixes #422

## Solution

Adds optional u64 `blockTimestamp` to `Log`.

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types/src/eth/log.rs [hunk 1]
```diff
@@ -14,6 +14,9 @@ pub struct Log<T = LogData> {
     /// Number of the block the transaction that emitted this log was mined in
     #[serde(with = "alloy_serde::u64_hex_opt")]
     pub block_number: Option<u64>,
+    /// The timestamp of the block.
+    #[serde(skip_serializing_if = "Option::is_none", with = "alloy_serde::u64_hex_opt", default)]
+    pub block_timestamp: Option<u64>,
```
Comment (mattsse) [inline crates/rpc-types/src/eth/log.rs:19]: can we reference the EIP for this here?

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic doc nit

File: crates/rpc-types/src/eth/log.rs (hunk 1 above)
Comment (cairoeth) [reply crates/rpc-types/src/eth/log.rs:19]: i dont think it's yet an EIP, so should i just link to [the proposal](https://ethereum-magicians.org/t/proposal-for-adding-blocktimestamp-to-logs-object-returned-by-eth-getlogs-and-related-requests)?

Comment (mattsse) [reply crates/rpc-types/src/eth/log.rs:19]: right then let's add this and https://github.com/ethereum/execution-apis/issues/295

Comment (cairoeth) [reply crates/rpc-types/src/eth/log.rs]: updated

**Review by mattsse — CHANGES_REQUESTED** [review]
could you please also include my commit from this branch:

https://github.com/alloy-rs/alloy/compare/matt/add-test?expand=1

I'm unable to to push here because this PR was opened from main

**Comment by cairoeth** [comment]
> could you please also include my commit from this branch:
> 
> https://github.com/alloy-rs/alloy/compare/matt/add-test?expand=1
> 
> I'm unable to to push here because this PR was opened from main

added

**Review by mattsse — APPROVED** [review]

**Pull request merged**